src/app/chat/[uuid]/page.tsx

```diff
@@ -1,25 +1,27 @@
 "use client";
 
 import { notFound, useParams } from "next/navigation";
+import { useEffect, useState } from "react";
 
+import { useChats } from "@/hooks/use-chats";
 import type { Chat } from "@/types/chat";
 
 export default function ConversationPage() {
   const parameters = useParams<{ uuid: string }>();
+  const { getChat } = useChats();
+  const [chat, setChat] = useState<Chat | null>(null);
 
-  // Get chats from local storage
-  const chats =
-    localStorage.getItem("chats") == null
-      ? []
-      : (JSON.parse(localStorage.getItem("chats")!) as Chat[]);
+  useEffect(() => {
+    const targetChat = getChat(parameters.uuid);
+    if (targetChat === null) {
+      notFound();
+    }
+    setChat(targetChat);
 
-  // Find chat by uuid
-  const chat = chats.find((c) => c.uuid === parameters.uuid);
+    if (!targetChat.recommendation) {
+      //TODO call api there
+    }
+  }, [parameters.uuid, getChat]);
```

Comment:
can we do something like this? I think it would be simpler to not introduce another state

```tsx
export default function ConversationPage() {
  const parameters = useParams<{ uuid: string }>();
  const { getChat, updateChat } = useChats();
  const chat = getChat(parameters.id);
  
  if (chat === null) {
    notFound();
  }
```

Reply:
After migrating to Jotai we can do this I think.

Additional state and `useEffect()` were to address strange behaviour which I don't really understand:
In short, after user refreshes the page (while being on /chat/{uuid}) the 404 is thrown in console, blank screen is displayed for a moment and then actual chat is displayed

Reply:
Probably because chat's page.tsx is a client component, and even those get rendered once on the server. Window API, from which local storage is from does not exist on the server. Server tries to run code which tries to access the local storage and fails - then the client takes over (hydration), local storage code re-runs because of a re-render and then it works

https://docs.solvro.pl/guides/web-development/common-mistakes/#jak-dzia%C5%82aj%C4%85client-components

therefore I think the above suggested approach wouldn't work